gradio/chat_interface.py

```diff
@@ -788,16 +800,18 @@ def _process_example(
         if self.multimodal:
             message = cast(ExampleMessage, message)
             if self.type == "tuples":
-                if "text" in message:
-                    result.append([message["text"], None])
                 for file in message.get("files", []):
                     result.append([file, None])
+                if "text" in message:
```

Comment:
For consistency with the non-caching examples case, we should add files to the history first and then the files.